src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java

```diff
@@ -46,57 +58,26 @@
     // TODO: test for proper exception on unsupported indexes (maybe via separate test?)
     // We have a 0.20.6.zip etc for this.
     
-    List<String> indexes = Arrays.asList(
-        "index-0.90.0.Beta1.zip",
-        "index-0.90.0.RC1.zip",
-        "index-0.90.0.RC2.zip",
-        "index-0.90.0.zip",
-        "index-0.90.1.zip",
-        "index-0.90.2.zip",
-        "index-0.90.3.zip",
-        "index-0.90.4.zip",
-        "index-0.90.5.zip",
-        "index-0.90.6.zip",
-        "index-0.90.7.zip",
-        "index-0.90.8.zip",
-        "index-0.90.9.zip",
-        "index-0.90.10.zip",
-        "index-0.90.11.zip",
-        "index-0.90.12.zip",
-        "index-0.90.13.zip",
-        "index-1.0.0.Beta1.zip",
-        "index-1.0.0.Beta2.zip",
-        "index-1.0.0.RC1.zip",
-        "index-1.0.0.RC2.zip",
-        "index-1.0.0.zip",
-        "index-1.0.1.zip",
-        "index-1.0.2.zip",
-        "index-1.0.3.zip",
-        "index-1.1.0.zip",
-        "index-1.1.1.zip",
-        "index-1.1.2.zip",
-        "index-1.2.0.zip",
-        "index-1.2.1.zip",
-        "index-1.2.2.zip",
-        "index-1.2.3.zip",
-        "index-1.2.4.zip",
-        "index-1.3.0.zip",
-        "index-1.3.1.zip",
-        "index-1.3.2.zip",
-        "index-1.3.3.zip",
-        "index-1.3.4.zip",
-        "index-1.3.5.zip",
-        "index-1.3.6.zip",
-        "index-1.3.7.zip",
-        "index-1.3.8.zip",
-        "index-1.3.9.zip",
-        "index-1.4.0.Beta1.zip",
-        "index-1.4.0.zip",
-        "index-1.4.1.zip",
-        "index-1.4.2.zip",
-        "index-1.4.3.zip",
-        "index-1.4.4.zip"
-    );
+    static List<String> indexes;
+    
+    @BeforeClass
+    public static void initIndexes() throws Exception {
+        indexes = new ArrayList<>();
+        URL dirUrl = OldIndexBackwardsCompatibilityTests.class.getResource(".");
+        Path dir = Paths.get(dirUrl.toURI());
+        DirectoryStream.Filter<Path> filter = new DirectoryStream.Filter<Path>() {
+            public boolean accept(Path file) throws IOException {
+                String filename = file.getFileName().toString();
+                return filename.startsWith("index-") && filename.endsWith(".zip");
```

Comment:
Can we add a comment, somewhere, that this test looks for all indexes matching the pattern index-*.zip?

I would maybe implement it this way too as a glob, with Files.newDirectoryStream(dir, "index-*.zip")